### PR TITLE
Uploader interface update

### PR DESCRIPTION
**Pull request by duttonw**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,8 +2,9 @@ language: python
 python:
     - "2.7"
 env:
-    - PGVERSION=9.1 CKAN_BRANCH=master
+    - PGVERSION=9.1 CKAN_BRANCH=qov-master
```
Comment (ThrawnCA) [inline .travis.yml:5]: Misspelled "qgov-master"

File: ckanext/s3filestore/uploader.py [hunk 2]
```diff
@@ -219,6 +222,46 @@ def upload(self, max_size=2):
...
+
+
+        key_path = self.filename()
+
+        if key_path is None:
+            log.warning('Key \'{0}\' not found in bucket \'{1}\''
+                     .format(key_path, self.bucket_name))
+
+        try:
+            # Small workaround to manage downloading of large files
```
Comment (ThrawnCA) [inline ckanext/s3filestore/uploader.py:249]: We could probably extract parts of this code to a helper function since it's now repeated three times IIUC.

**Comment by ThrawnCA** [comment]
Still misspelling the CKAN branch name.

File: ckanext/s3filestore/plugin.py [hunk 3]
```diff
@@ -60,12 +60,14 @@ def get_uploader(self, upload_to, old_filename=None):
...
         with SubMapper(map, controller='ckanext.s3filestore.controller:S3Controller') as m:
             # Override the resource download links
-            m.connect('resource_download',
-                      '/dataset/{id}/resource/{resource_id}/download',
-                      action='resource_download')
-            m.connect('resource_download',
-                      '/dataset/{id}/resource/{resource_id}/download/{filename}',
-                      action='resource_download')
+            # if plugins.toolkit.check_ckan_version(max_version='2.8.2') & \
+            if hasattr(DefaultResourceUpload, "download") == False :
```
Comment (ThrawnCA) [inline ckanext/s3filestore/plugin.py:64]: How is the download URL routed when the `download` function *does* exist?

Comment (duttonw) [reply ckanext/s3filestore/plugin.py:64]: ckan/controllers/package.py 
is changed to
return upload.download(rsc['id'], filename)
from
resource download bits.

**Pull request merged**